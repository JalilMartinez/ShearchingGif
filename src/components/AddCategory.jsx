import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCat}) => {

    const [input, setInput ] = useState('')
    const OnInputChange = (event)=>{
       
        setInput(event.target.value);
        console.log(event.target.value);
    }
    const onSubmit = (evento) =>{
        event.preventDefault(); 
        if(input.trim().length<=1) return;
        console.log(input);
        
        
        setInput('');
        // setCate(cat => [input,...cat]);   
        onNewCat(input.trim());
    }
    return(
        <>
            <form onSubmit={onSubmit} aria-label='form'>
                <input
                    type='text'
                    placeholder='buscar gifts'
                    value={input}
                    onChange={OnInputChange}
                    />      
            </form>
            <button onClick={onSubmit}>Agregar</button>
            
        </>
    )

}

AddCategory.propTypes={
    onNewCat: PropTypes.func.isRequired,
}