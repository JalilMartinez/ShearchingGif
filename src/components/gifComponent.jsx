import PropTypes from 'prop-types';
import React from "react";


export const GifComponent = ( {title, url, id} ) => {
    
    return(
        <div className="card">
            
            <img key={id} src={url} alt={title} />
            <p>{title}</p>
        </div>
        
    )
}

GifComponent.prototypes = {
    title: PropTypes.string.isRequire,
    url: PropTypes.string.isRequire,
}