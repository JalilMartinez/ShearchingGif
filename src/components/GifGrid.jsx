import React from "react";


import { GifComponent } from "./gifComponent";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GifGrid= ({cat})=>{
    const {images, isLoading} = useFetchGifs(cat); //custom hook
    
    
    return(
        <>
            <h3>{cat}</h3>
            
                {
                    // isLoading ? <h2>Cargando...</h2>: null
                    isLoading && <h2>Cargando...</h2>
                }
            
            <ol className="card-grid">
                {
                    images.map(ima =>{
                       return <GifComponent 
                             key={ima.id}
                             {...ima}/> 
                    })
                }
            </ol>
            
                        
                    
        </>
    )
}

GifGrid.propTypes = {
    cat: PropTypes.string.isRequired,
}