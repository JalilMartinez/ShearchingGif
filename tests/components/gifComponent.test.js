import { GifComponent } from "../../src/components/gifComponent";
import {render, screen} from '@testing-library/react';
describe('Pruebas en el nombre del componente', ()=>{
    const title = 'Evangelion';
    const url = 'https://evangelion-upon.com/asuka.png';
    test('debe de hacer match con el snapshot', ()=>{
        const {container} = render(<GifComponent title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });
    test('debe de mostrar la imagen con el url y el atl indicado', ()=>{
         render(<GifComponent url={url} title={title}/>);
        // expect(screen.getByRole('img').getAttribute('src')).toEqual(url);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });
    test('Debe de mostrar ekl titulo en el componente', ()=>{
        render(<GifComponent url={url} title={title}/>);
        expect(screen.getByText(title));
    });
});
