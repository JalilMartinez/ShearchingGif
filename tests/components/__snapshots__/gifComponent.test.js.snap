// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el nombre del componente debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Evangelion"
      src="https://evangelion-upon.com/asuka.png"
    />
    <p>
      Evangelion
    </p>
  </div>
</div>
`;
